src/service/viewport/viewport-impl.js

```diff
@@ -204,6 +204,14 @@ export class ViewportImpl {
         // Ignore errors.
       }
     }
+
+    // BF-cache navigation sometimes breaks clicks in an iframe on iOS. See
+    // https://github.com/ampproject/amphtml/issues/30838 for more details.
+    // The solution is to make a "fake" scrolling API call.
+    const isIframedIos = Services.platformFor(win).isIos() && isIframed(win);
```

Comment:
Does this run when loading from bf-cache? LGTM assuming it does.

Reply:
Yes. Definitely.